_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,349 @@
+
+```java
+public interface TagSearchRepository extends ReactiveElasticsearchRepository<Tag, String>, TagSearchRepositoryInternal {}
+
+interface TagSearchRepositoryInternal {
+    Flux<Tag> search(String query, Pageable pageable);
+}
+
+class TagSearchRepositoryInternalImpl implements TagSearchRepositoryInternal {
+
+    private final ReactiveElasticsearchTemplate reactiveElasticsearchTemplate;
+
+    TagSearchRepositoryInternalImpl(ReactiveElasticsearchTemplate reactiveElasticsearchTemplate) {
+        this.reactiveElasticsearchTemplate = reactiveElasticsearchTemplate;
+    }
+
+    @Override
+    public Flux<Tag> search(String query, Pageable pageable) {
+        List<FieldSortBuilder> builders = new SortToFieldSortBuilderConverter().convert(pageable.getSort());
+
+        NativeSearchQueryBuilder queryBuilder = new NativeSearchQueryBuilder()
+            .withQuery(queryStringQuery(query))
+            .withPageable(PageRequest.of(pageable.getPageNumber(), pageable.getPageSize()));
+
+        builders
+            .stream()
+            .forEach(builder -> {
+                queryBuilder.withSort(builder);
+            });
+
+        NativeSearchQuery nativeSearchQuery = queryBuilder.build();
+        return reactiveElasticsearchTemplate.search(nativeSearchQuery, Tag.class).map(SearchHit::getContent);
+    }
+}
+```
+
+_Autoconfiguration_<br>
+
+`ReactiveElasticsearchRestClientAutoConfiguration` configures the reactive Elasticsearch client from the Spring Data Elasticsearch properties. `ElasticsearchDataAutoConfiguration` is the root configuration class that triggers the reactive search template initialization.
+
+## Learn more about Spring Data Elasticsearch and Okta
+
+JHipster helps to simplify the setup of Spring Boot applications or microservices with search capabilities. I hope you enjoyed this quick introduction to Elasticsearh integration options, and could taste the advantages of the Spring Data repository abstraction for encapsulating some basic Elasticsearch operations. Remember to check the [compatiblity matrix](https://docs.spring.io/spring-data/elasticsearch/docs/current/reference/html/#preface.versions) to match the right Spring Data Elasticsearch dependency for your Elasticsearch version. Keep learning, and for more examples and recipes on Elasticsearch and Okta integrations for Spring Boot, check out the following links:
+
+- [Elasticsearch Clients](https://www.elastic.co/guide/en/elasticsearch/client/index.html)
+- [Elasticsearch Mapping Multi-Fields](https://www.elastic.co/guide/en/elasticsearch/reference/current/multi-fields.html)
+- [Get Started with the ELK Stack](/blog/2019/09/26/get-started-elk-stack)
+- [Spring Boot and Okta in 2 Minutes](/blog/2020/11/24/spring-boot-okta)
+- [Reactive Java Microservices with Spring Boot and JHipster](/blog/2021/01/20/reactive-java-microservices)
+- [Elasticsearh REST APIs](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html)
```

Comment:
```suggestion
- [Elasticsearch REST APIs](https://www.elastic.co/guide/en/elasticsearch/reference/current/rest-apis.html)
```

More JHipster!

Reply:
It won't let me comment from below, so I'll try from above. I'd recommend changing the line below to:

> You can find the source code of this tutorial on GitHub in the [okta-word1-word2-example](URL) repo. 

Also needs "follow us" links for Twitter, LinkedIn, and YouTube.